File: LayoutTests/http/tests/security/contentSecurityPolicy/resources/echo-report.php
<?php
require_once "report-file-path.php";

while (!file_exists($reportFilePath)) {
    usleep(10000);
    // file_exists() caches results, we want to invalidate the cache.
    clearstatcache();
}

echo "<html><body>\n";
echo "CSP report received:";
$reportFile = fopen($reportFilePath, 'r');
while ($line = fgets($reportFile)) {
    echo "<br>";
    echo trim($line);
}
fclose($reportFile);
unlink($reportFilePath);
echo "<script>";
echo "if (window.testRunner)";
echo "    testRunner.notifyDone();";
echo "</script>";
echo "</body></html>";
?>

File: LayoutTests/http/tests/security/contentSecurityPolicy/resources/echo-report.py
#!/usr/bin/env python3

import os
import sys
import time
from report_file_path import report_filepath

while not os.path.isfile(report_filepath):
    time.sleep(0.01)

sys.stdout.write(
    'Content-Type: text/html\r\n\r\n'
    '<html><body>\n'
    'CSP report received:'
)

report_file = open(report_filepath, 'r')
for line in report_file.readlines():
    sys.stdout.write('<br>{}'.format(line.strip()))

report_file.close()
if os.path.isfile(report_filepath):
    os.remove(report_filepath)

sys.stdout.write(
    '<script>'
    'if (window.testRunner)'
    '    testRunner.notifyDone();'
    '</script>'
    '</body></html>'
)
